bot.go

```diff
@@ -93,14 +101,14 @@ func (b *Bot) Addplayer(s *discordgo.Session, m *discordgo.MessageCreate, name s
 			if _, ok := b.games[g]; !ok {
 				b.games[g] = Game{Players: make(map[string]PlayerMetadata), Blue: make(map[string]bool), Red: make(map[string]bool), mutex: new(sync.Mutex), RedCaptain: new(string), BlueCaptain: new(string)}
 			}
-			b.games[g].mutex.Lock()
-			defer b.games[g].mutex.Unlock()
-
 			if len(b.games[g].Players)+len(b.games[g].Red)+len(b.games[g].Blue) == mod.MaxPlayers {
 				return
 			}
+
+			b.games[g].mutex.Lock()
+
```

Comment:
I would keep the defer, otherwise we have to make sure we unlock in all branches?

Reply:
Yeah idk how to handle this nicely. I'm manually unlocking due to a deadlock when calling `b.List` from this function. Might have unintended consequences to unlock in defer on top of that